subprojects/core/src/main/java/org/gradle/initialization/DefaultParallelConfiguration.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.initialization;
+
+import org.gradle.api.Incubating;
+
+import java.io.Serializable;
+
+public class DefaultParallelConfiguration implements Serializable, ParallelConfiguration {
+    private boolean parallelProjectExecution;
+    private int maxWorkerCount;
+
+    public DefaultParallelConfiguration() {
+        maxWorkerCount = Runtime.getRuntime().availableProcessors();
+    }
+
+    /**
+     * Returns true if parallel project execution is enabled.
+     *
+     * @see #getMaxWorkerCount()
+     */
+    @Incubating
+    public boolean isParallelProjectExecutionEnabled() {
+        return parallelProjectExecution;
+    }
+
+    /**
+     * Enables/disables parallel project execution.
+     *
+     * @see #isParallelProjectExecutionEnabled()
+     */
+    @Incubating
```

Comment:
I don't think the goal here is to make the class internal as it wasn't the case before. We should shelve this for another PR.